base/array.hpp

```diff
@@ -55,6 +56,36 @@ struct UniqueArray final {
   std::int64_t size;  // In number of elements.
 };
 
+// A simple container for an array and a size.  The client is expected to use
+// aggregate initialization for this type, and to ensure that the values passed
+// for |data| and |size| are consistent.  This type is *not* self-initializing.
```

Comment:
Make an issue or add a TODO to investigate whether we could have a constructor from a parameter pack (that forwards the parameters to aggregate initialization in some fashion), since the `{{elements}, size}` initalization is error-prone.